### PR TITLE
chore(db pull): log how many embedded documents were introspected

### LINKED ISSUE #11827: MongoDB Introspection does not output how many embedded documents were introspected
```
C:\Users\Jan\Documents\throwaway\migrationsTeam319 (main -> origin) (migrationsteam319@1.0.0)
λ npx prisma db pull --url mongodb+srv://user:password@cluster0.tagd6.mongodb.net/sample_analytics --force --composite-type-depth 2
Prisma schema loaded from prisma\schema.prisma
Environment variables loaded from .env
Datasource "db"

Introspecting …

✔ Introspected 3 models and wrote them into prisma\schema.prisma in 125.03s

Run prisma generate to generate Prisma Client.
```

but resulting schema file has 458x `type`:
```
C:\Users\Jan\Documents\throwaway\migrationsTeam319 (main -> origin) (migrationsteam319@1.0.0)
λ find /C "type" "prisma/schema.prisma"

---------- PRISMA/SCHEMA.PRISMA: 458
```

### PR DESCRIPTION
**Pull request by Jolg42**
Closes https://github.com/prisma/prisma/issues/11827

### CODE REVIEW

**Comment by Jolg42** [comment]
@janpio note that this logging is not specific to MongoDB, just matching `type` like we were doing previously with models.

So we might need to revisit / improve when composite types land in SQL connectors

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 1]
```diff
@@ -589,13 +589,38 @@ describeIf(process.platform !== 'win32' && !isMacOrWindowsCI)('MongoDB', () => {
     expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
     expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(`
 
-      // *** WARNING ***
-      // 
-      // The following fields had data stored in multiple types. The most common type was chosen. If loading data with a type that does not match the one in the data model, the client will crash. Please see the issue: https://github.com/prisma/prisma/issues/9654
-      // - Model "users", field: "numberOrString1", chosen data type: "Int32"
-      // - Type "UsersHobbies", field: "numberOrString2", chosen data type: "Int32"
-      // 
+                  // *** WARNING ***
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:592]: Why are we not getting the same output in this test as in the one below?

Comment (Jolg42) [reply packages/migrate/src/__tests__/DbPull.test.ts:592]: this is using --print, below is using --force

**Review by janpio — APPROVED** [review]
(The indentation changes in the snapshots... *sigh)

Looks good!

**Pull request merged**